nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/util/S4JClientConfConverter.java

```diff
@@ -81,15 +81,14 @@ public static Map<String, Object> convertRawProducerConf(Map<String, String> pul
             if (StringUtils.equalsAnyIgnoreCase(confVal, "LZ4", "ZLIB", "ZSTD", "SNAPPY")) {
                 CompressionType compressionType = CompressionType.NONE;
 
-                switch (StringUtils.upperCase(confVal)) {
-                    case "LZ4":
-                        compressionType = CompressionType.LZ4;
-                    case "ZLIB":
-                        compressionType = CompressionType.ZLIB;
-                    case "ZSTD":
-                        compressionType = CompressionType.ZSTD;
-                    case "SNAPPY":
-                        compressionType = CompressionType.SNAPPY;
+                if (StringUtils.equalsIgnoreCase(confVal, "LZ4")) {
```

Comment:
This code could be much cleaner with an enum, _enum_.valueOf(...) and a newer switch form.

Reply:
Fixed as suggested